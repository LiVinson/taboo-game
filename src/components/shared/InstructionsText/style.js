import styled from "styled-components"

export const StyledInstructionsText = styled.p`
  font-size: 2rem;
  margin-bottom: 2rem;
  padding: 0 1rem;
  text-align:${props => props.align ? props.align : "left"};
  color: ${props => props.theme.color.grayDark2};
  &:not(:last-child){
        margin-bottom:1rem;
    }
`

