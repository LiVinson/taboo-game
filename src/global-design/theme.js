const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
