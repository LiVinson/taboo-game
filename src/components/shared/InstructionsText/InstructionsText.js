import React from 'react'
import PropTypes from 'prop-types'
import { StyledInstructionsText } from './style.js'

export const InstructionsText = ({ children, align }) => {
	return <StyledInstructionsText align={align}>{children}</StyledInstructionsText>
}

InstructionsText.propTypes = {
	children: PropTypes.node.isRequired,
}
