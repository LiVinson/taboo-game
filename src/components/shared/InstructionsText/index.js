import { InstructionsText } from "./InstructionsText"

export default InstructionsText