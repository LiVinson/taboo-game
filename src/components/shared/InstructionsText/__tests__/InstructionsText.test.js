import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'
import theme from '../../../../global-design/theme'
import {InstructionsText} from "../InstructionsText"

test("InstructionsText renders correctly", () => {
    const text = "Here are some instructions to display."
    const wrapper = renderer
    .create(
        <ThemeProvider theme={theme}>
            <InstructionsText theme={theme}>
                {text}
            </InstructionsText>
        </ThemeProvider>
    )
    .toJSON()

expect(wrapper).toMatchSnapshot()
})

test("InstructionsText renders text passed as children", () => {
    const text = "Here are some instructions to display."
    const wrapper = shallow(
            <InstructionsText theme={theme}>
                {text}
            </InstructionsText>
    )
 
    expect(wrapper.text()).toEqual(text)      
})