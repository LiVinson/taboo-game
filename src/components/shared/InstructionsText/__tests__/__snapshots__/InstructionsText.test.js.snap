// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructionsText renders correctly 1`] = `
.c0 {
  font-size: 2rem;
  margin-bottom: 2rem;
  padding: 0 1rem;
  text-align: left;
  color: #333333;
}

.c0:not(:last-child) {
  margin-bottom: 1rem;
}

<p
  className="c0"
>
  Here are some instructions to display.
</p>
`;
